sdk/resourcemanager/azure-resourcemanager-containerservice/src/main/java/com/azure/resourcemanager/containerservice/models/KubernetesCluster.java

```diff
@@ -707,6 +759,22 @@ interface WithNetworkProfile {
              * @return the next stage of the update
              */
             Update withNetworkProfile(ContainerServiceNetworkProfile networkProfile);
+
+            /**
+             * Updates the cluster's network policy.
+             *
+             * @param networkPolicy the cluster's networkPolicy
+             * @return the next stage of the update
+             */
+            Update withNetworkPolicy(NetworkPolicy networkPolicy);
+
+            /**
+             * Updates the cluster's network data plam.
+             *
+             * @param networkDataPlan the cluster's networkDataPlan
+             * @return the next stage of the update
+             */
+            Update withNetworkDataPlan(NetworkDataplane networkDataPlan);
```

Comment:
Sorry, my fault.

We already had `withNetworkProfile(ContainerServiceNetworkProfile networkProfile);`. Please remove this 2 (and the impl in KubernetesClusterImpl.java)

Please keep the test (use `withNetworkProfile`) to set them (I didn't see update on `networkMode`, it there a reason?).

Reply:
1. `NetworkMode` and `NetworkPluginMode` cannot be specified if `NetworkPlugin` is anything other than `azure`.
2. `NetworkMode` and `NetworkPlugin` are important properties for create `aks`, so they can only be update by redeploy.